Looms.PoS/Looms.PoS.Application/Features/GiftCard/Commands/CreateGiftCard/CreateGiftCardCommandHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Looms.PoS.Application.Interfaces;
+using Looms.PoS.Application.Interfaces.ModelsResolvers;
+using Looms.PoS.Application.Models.Requests.GiftCard;
+using Looms.PoS.Domain.Interfaces;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Looms.PoS.Application.Features.GiftCard.Commands.CreateGiftCard;
+
+public class CreateGiftCardCommandHandler : IRequestHandler<CreateGiftCardCommand, IActionResult>
+{
+    private readonly IGiftCardsRepository _giftCardsRepository;
+    private readonly IHttpContentResolver _httpContentResolver;
+    private readonly IGiftCardModelsResolver _modelsResolver;
+
+    public CreateGiftCardCommandHandler(
+        IGiftCardsRepository giftCardsRepository,
+        IHttpContentResolver httpContentResolver,
+        IGiftCardModelsResolver modelsResolver)
+    {
+        _giftCardsRepository = giftCardsRepository;
+        _httpContentResolver = httpContentResolver;
+        _modelsResolver = modelsResolver;
+    }
+
+    public async Task<IActionResult> Handle(CreateGiftCardCommand command, CancellationToken cancellationToken)
+    {
+        var giftCardRequest = await _httpContentResolver.GetPayloadAsync<CreateGiftCardRequest>(command.Request);
+
+        // TODO: add businessId
```

Comment:
why not add it now?

Reply:
I think we should send business id through header and then have a middleware to validate it. If so, we could easily scope everything to selected business.

Reply:
not a bad idea